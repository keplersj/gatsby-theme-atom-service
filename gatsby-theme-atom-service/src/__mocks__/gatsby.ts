import * as React from "react";
const gatsby = jest.requireActual("gatsby");

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest.fn().mockImplementation(
    ({
      // these props are invalid for an `a` tag
      /* eslint-disable @typescript-eslint/no-unused-vars */
      /* eslint-disable no-unused-vars */
      activeClassName,
      activeStyle,
      getProps,
      innerRef,
      partiallyActive,
      ref,
      replace,
      /* eslint-enable @typescript-eslint/no-unused-vars */
      /* eslint-enable no-unused-vars */
      to,
      ...rest
    }): React.ReactElement =>
      React.createElement("a", {
        ...rest,
        href: to
      })
  ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn()
};
