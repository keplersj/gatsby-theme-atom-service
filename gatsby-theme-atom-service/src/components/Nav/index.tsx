import { graphql, useStaticQuery } from "gatsby";
import * as React from "react";
import {
  AppBar,
  Toolbar,
  IconButton,
  Button,
  Typography,
  Drawer,
  List,
  Divider,
  ListItem,
  ListItemText,
  Hidden
} from "@material-ui/core";
import { Menu as MenuIcon } from "@material-ui/icons";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1
  },
  bar: {
    backgroundColor: "#66595c"
  },
  menuButton: {
    marginRight: theme.spacing(2)
  },
  title: {
    flexGrow: 1
  }
}));

interface NavQuery {
  site: {
    siteMetadata: {
      title: string;
      nav: {
        name: string;
        url: string;
      }[];
    };
  };
  allProvidersYaml: {
    edges: {
      node: {
        id: string;
        title: string;
        modal: string;
      };
    }[];
  };
}

const Nav = (): React.ReactElement => {
  const classes = useStyles({});
  const [state, setState] = React.useState({
    drawerOpen: false
  });
  const data = useStaticQuery<NavQuery>(graphql`
    query NavQuery {
      site {
        siteMetadata {
          title
          nav {
            name
            url
          }
        }
      }
      allProvidersYaml {
        edges {
          node {
            id
            title
            modal
          }
        }
      }
    }
  `);

  // Disabling this lint rule for now, it does raise a good point for future refactoring though.
  // TODO: refactor
  // eslint-disable-next-line unicorn/consistent-function-scoping
  const toggleDrawer = (open: boolean) => (
    event: React.KeyboardEvent | React.MouseEvent
  ): void => {
    if (
      event.type === "keydown" &&
      ((event as React.KeyboardEvent).key === "Tab" ||
        (event as React.KeyboardEvent).key === "Shift")
    ) {
      return;
    }

    setState({
      ...state,
      drawerOpen: open
    });
  };

  return (
    <>
      <nav className={classes.root}>
        <AppBar position="static" className={classes.bar}>
          <Toolbar>
            <IconButton
              edge="start"
              color="inherit"
              aria-label="menu"
              className={classes.menuButton}
              onClick={toggleDrawer(true)}
            >
              <MenuIcon />
            </IconButton>
            <Typography variant="h6" className={classes.title}>
              {data.site.siteMetadata.title}
            </Typography>
            <Hidden only={["xs", "sm"]}>
              {data.site.siteMetadata.nav.map(link => (
                <Button key={link.name} color="inherit" href={link.url}>
                  {link.name}
                </Button>
              ))}
            </Hidden>
          </Toolbar>
        </AppBar>
      </nav>
      <Drawer open={state.drawerOpen} onClose={toggleDrawer(false)}>
        <div
          role="presentation"
          data-testid="drawer-content"
          onClick={toggleDrawer(false)}
          onKeyDown={toggleDrawer(false)}
        >
          <List>
            {data.site.siteMetadata.nav.map(link => (
              <ListItem button component="a" key={link.name} href={link.url}>
                <ListItemText primary={link.name} />
              </ListItem>
            ))}
          </List>
          <Divider />
          <List>
            {data.allProvidersYaml.edges.map(({ node }) => (
              <ListItem
                button
                component="a"
                key={node.id}
                href={`#${node.modal}`}
              >
                <ListItemText primary={node.title} />
              </ListItem>
            ))}
          </List>
        </div>
      </Drawer>
    </>
  );
};

export default Nav;
