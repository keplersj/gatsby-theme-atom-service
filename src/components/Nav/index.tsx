import { graphql, useStaticQuery } from "gatsby";
import * as React from "react";
import {
  AppBar,
  Toolbar,
  IconButton,
  Button,
  Typography,
  Drawer,
  List,
  Divider,
  ListItem,
  ListItemText,
  Hidden
} from "@material-ui/core";
import { Menu as MenuIcon } from "@material-ui/icons";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1
  },
  bar: {
    backgroundColor: "#66595c"
  },
  menuButton: {
    marginRight: theme.spacing(2)
  },
  title: {
    flexGrow: 1
  }
}));

interface NavQuery {
  site: {
    siteMetadata: {
      title: string;
      nav: {
        name: string;
        url: string;
      }[];
    };
  };
  allProvidersYaml: {
    edges: {
      node: {
        id: string;
        title: string;
        modal: string;
      };
    }[];
  };
}

const Nav = () => {
  const classes = useStyles({});
  const [state, setState] = React.useState({
    drawerOpen: false
  });
  const data = useStaticQuery<NavQuery>(graphql`
    query NavQuery {
      site {
        siteMetadata {
          title
          nav {
            name
            url
          }
        }
      }
      allProvidersYaml {
        edges {
          node {
            id
            title
            modal
          }
        }
      }
    }
  `);

  const toggleDrawer = (open: boolean) => (
    event: React.KeyboardEvent | React.MouseEvent
  ) => {
    if (
      event.type === "keydown" &&
      ((event as React.KeyboardEvent).key === "Tab" ||
        (event as React.KeyboardEvent).key === "Shift")
    ) {
      return;
    }

    setState({ ...state, drawerOpen: open });
  };

  return (
    <>
      <nav className={classes.root}>
        <AppBar position="static" className={classes.bar}>
          <Toolbar>
            <IconButton
              edge="start"
              color="inherit"
              aria-label="menu"
              className={classes.menuButton}
              onClick={toggleDrawer(true)}
            >
              <MenuIcon />
            </IconButton>
            <Typography variant="h6" className={classes.title}>
              {data.site.siteMetadata.title}
            </Typography>
            <Hidden only={["xs", "sm"]}>
              {data.site.siteMetadata.nav.map(link => (
                <Button color="inherit" href={link.url}>
                  {link.name}
                </Button>
              ))}
            </Hidden>
          </Toolbar>
        </AppBar>
      </nav>
      <Drawer open={state.drawerOpen} onClose={toggleDrawer(false)}>
        <div
          role="presentation"
          data-testid="drawer-content"
          onClick={toggleDrawer(false)}
          onKeyDown={toggleDrawer(false)}
        >
          <List>
            {data.site.siteMetadata.nav.map(link => (
              <ListItem button component="a" key={link.name} href={link.url}>
                <ListItemText primary={link.name} />
              </ListItem>
            ))}
          </List>
          <Divider />
          <List>
            {data.allProvidersYaml.edges.map(({ node }) => (
              <ListItem
                button
                component="a"
                key={node.id}
                href={`#${node.modal}`}
              >
                <ListItemText primary={node.title} />
              </ListItem>
            ))}
          </List>
        </div>
      </Drawer>
    </>
  );
};

export default Nav;
