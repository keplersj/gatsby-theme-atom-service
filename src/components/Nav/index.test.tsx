import * as React from "react";
import renderer from "react-test-renderer";
import { useStaticQuery } from "gatsby";
import Nav from ".";

describe("Nav", () => {
  beforeEach(() => {
    (useStaticQuery as jest.Mock).mockImplementationOnce(() => ({
      site: {
        siteMetadata: {
          title: "Example Site",
          nav: [
            {
              name: "Example Link",
              url: "https://example.dev/"
            }
          ]
        }
      },
      allProvidersYaml: {
        edges: [
          {
            node: {
              id: "123",
              title: "Example Section",
              modal: "example"
            }
          }
        ]
      }
    }));
  });

  it("renders as expected", () => {
    const tree = renderer.create(<Nav />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
