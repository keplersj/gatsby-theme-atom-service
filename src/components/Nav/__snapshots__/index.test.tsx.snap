// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav opens the drawer when the menu icon is pressed 1`] = `
<div
  data-testid="drawer-content"
  role="presentation"
>
  <ul
    class="MuiList-root MuiList-padding"
  >
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      href="https://example.dev/"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
        >
          Example Link
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </ul>
  <hr
    class="MuiDivider-root"
  />
  <ul
    class="MuiList-root MuiList-padding"
  >
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      href="#example"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
        >
          Example Section
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </ul>
</div>
`;

exports[`Nav renders as expected 1`] = `
<nav
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic makeStyles-bar-2 MuiAppBar-colorPrimary"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-3 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <h6
        className="MuiTypography-root makeStyles-title-4 MuiTypography-h6"
      >
        Example Site
      </h6>
    </div>
  </header>
</nav>
`;
